trieste/models/normalization.py

```diff
@@ -0,0 +1,350 @@
+        """
+        return variance * self._std ** 2
+
+
+class MinMaxTransformer(DataTransformer):
+    """Transforms data to the unit cube, i.e. to the interval [0, 1], for all dimensions."""
+
+    def __init__(self, data: TensorType) -> None:
+        """Initialize a transformer that normalizes data to the unit cube.
+
+        :param data: The unnormalized data [N, D].
+        """
+        super().__init__(data)
+
+    def set_parameters(self, data: TensorType) -> None:
+        """Set parameters of the transform; the minimum (self._min) and the difference between the
+        maximum and the minimum (self._delta) of the data along each dimension.
+
+        :param data: The unnormalized data [N, D].
+        """
+        self._min = tf.squeeze(tf.math.reduce_min(data, axis=0))
+        max_data = tf.squeeze(tf.math.reduce_max(data, axis=0))
+        self._delta = max_data - self._min
+
+    def transform(self, data: TensorType) -> TensorType:
+        """Normalize data.
+
+        :param data: The unnormalized data. Shape must be [..., D] if transform operates over
+        multiple dimensions else any shape.
+        :return: The normalized data. Same shape as `data`.
+        """
+        return (data - self._min) / self._delta
+
+    def inverse_transform(self, data: TensorType) -> TensorType:
+        """Transform normalized data to unnormalized space.
+
+        :param data: The normalized data. Shape [..., D] if transform operates over multiple
+        dimensions else any shape.
+        :return: The unnormalized data. Same shape as `data`.
+        """
+        return data * self._delta + self._min
+
+    def transform_variance(self, variance: TensorType) -> TensorType:
+        """Transform the variance of unnormalized data to the variance of normalized data.
+
+        :param variance: The variance of the unnormalized data. Shape [..., D] if transform
+        operates over multiple dimensions else any shape.
+        :return: The variance of the normalized data. Same shape as `data`.
+        """
+        return variance / self._delta ** 2
+
+    def inverse_transform_variance(self, variance: TensorType) -> TensorType:
+        """Transform the variance of normalized data to the variance of unnormalized data.
+
+        :param variance: The variance of the unnormalized data. Shape [..., D] if transform
+        operates over multiple dimensions else any shape.
+        :return: The variance of the normalized data. Same shape as `data`.
+        """
+        return variance * self._delta ** 2
+
+
+class DataTransformModelWrapper(TrainableProbabilisticModel):
+    """
+    A wrapper that handles data transformations for :class: `TrainableProbabilisticModel`s.
+
+    Usage requires creating a new class that inherits from this class and the desired model, e.g.
+    ```
+    class GPRwithDataNormalization(DataTransformModelWrapper, GaussianProcessRegression):
```

Comment:
I commented in the notebook, lets provide these simple new objects in model specific directory

Reply:
Thanks for the comments! I have addressed most of them but there are still a few minor points that need further discussion